source/ceylon/net/http/server/MatcherRule.ceylon

```diff
@@ -16,6 +17,22 @@ shared abstract class Matcher() {
             => Or(this, other);
 }
 
+"Matcher to leverage Undertow's template mechanism for path templates and path parameters.
+ It should be given to an Endpoint without combining it with other matchers.
+ 
```

Comment:
To prevent combining it with other matchers can we solve it with union type of Endpoint path parameter?
By setting a path parameter to be `Matcher|TemplateMatcher`, then `TemplateMatcher` doesn't need to extend Matcher and may not provide `and` and `or` capabilities.